perceval/components/source.py

```diff
@@ -28,58 +28,93 @@
+        :param brightness: the brightness of the source defined the probability per laser pulse to collect >1 photon at
+            the input of the circuit
+        :param multiphoton_component: second order intensity autocorrelation at zero time delay :math:`g^{(2)}(0)`
+        :param multiphoton_model: `distinguishable` if additional photons are distinguishable, `indistinguishable`
+          otherwise
+        :param purity: preserved for back-compatibility if multiphoton_model is not set. :math:`purity = 1-g^{(2)}(0)`
         :param indistinguishability: indistinguishability parameter as defined by `indistinguishability_model`
-        :param indistinguishability_model: `homv` defines indistinguishability as HOM visibility, `linear` defines
-            indistinguishability as ratio of indistinguishable photons
+        :param indistinguishability_model: `homv` defines indistinguishability as 2-photon wavepacket overlap,
+            `linear` defines indistinguishability as ratio of indistinguishable photons
+        :param overall_transmission: transmission of the optical system.
         :param context: gives a local context for source specific features, like `discernability_tag`
         """
+        if multiphoton_component is None:
+            if purity is None:
+                multiphoton_component = 0
+            else:
+                multiphoton_component = 1 - purity
+        else:
+            assert purity is None, "cannot set both purity and multiphoton_component"
         self.brightness = brightness
-        self.purity = purity
-        self._purity_model = purity_model
+        self.overall_transmission = overall_transmission
+        self.multiphoton_component = multiphoton_component
+        self._multiphoton_model = multiphoton_model
+        assert self._multiphoton_model in ["distinguishable", "indistinguishable"], "invalid value for purity_model"
         self.indistinguishability = indistinguishability
         self._indistinguishability_model = indistinguishability_model
         assert self._indistinguishability_model in ["homv", "linear"], "invalid value for indistinguishability_model"
         self._context = context or {}
+        # discernability_tag starts at 2. Label 0 is the single-photon. Label 1 is the distinguishable noise photon.
         if "discernability_tag" not in self._context:
-            self._context["discernability_tag"] = 1
+            self._context["discernability_tag"] = 2
 
     def probability_distribution(self):
         r"""returns SVDistribution on 1 mode associated to the source
         """
+            # g2 = 2p2/(p1+2p2)**2
+            # p1 + p2 = beta
+            # p1 + 2*p2 = mu
+
+        g2 = self.multiphoton_component
+        mu = self.brightness
+        eta = self.overall_transmission
+
+        beta = mu - g2 * mu ** 2 / 2
+        p2 = g2 * mu ** 2 / 2
+        p1 = beta - p2
+
         svd = SVDistribution()
-        if self.brightness != 1:
```

Comment:
did you remove this on purpose? without this, the Source is not generating `[0]` states and tests in processor with several source fail

Reply:
Yes! Because i) we will always postselect on detected events and ii) the losses and the brightness of the sources are already taken into accounts with mu and eta.